src/components/Header/CKBNodeComp/CKBNodeModal.tsx

```diff
@@ -1,39 +1,171 @@
+  }>
+> = ({ children, initialData, onSubmit }) => {
+  const [alias, setAlias] = useState(initialData?.alias ?? '')
+  const [url, setUrl] = useState(initialData?.url ?? '')
+
+  return (
+    <form
+      className={styles.nodeEditForm}
+      onSubmit={e => {
+        e.preventDefault()
+        e.stopPropagation()
+        onSubmit({ alias, url })
+      }}
+    >
+      <div>
+        <label>Alias</label>
+        <input value={alias} onChange={e => setAlias(e.target.value)} />
+      </div>
+
+      <div>
+        <label>RPC URL</label>
+        <input value={url} onChange={e => setUrl(e.target.value)} />
+      </div>
+
+      {children}
+    </form>
+  )
+}
+
 export const CKBNodeModal = ({ onClose }: { onClose: () => void }) => {
   const { t } = useTranslation()
   const ref = useRef<HTMLDivElement>(null)
-  const { isActivated, setIsActivated } = useCKBNode()
+  const { isActivated, setIsActivated, nodeService, nodes, addNode, removeNode, switchNode, editNode } = useCKBNode()
+  const [panel, setPanel] = useState<'main' | 'add' | 'edit'>('main')
+  const [editIndex, setEditIndex] = useState(0)
+
+  const NodeEditPanel = () => (
+    <div className={styles.contentWrapper}>
+      <div className={styles.modalTitle}>
+        <button type="button" onClick={() => setPanel('main')} className={styles.closeBtn}>
+          <img src={BackIcon} alt="back icon" />
+        </button>
+
+        <p>{t('node.modify_node')}</p>
+        <button type="button" onClick={onClose} className={styles.closeBtn}>
+          <img src={CloseIcon} alt="close icon" />
+        </button>
+      </div>
+
+      <NodeEditForm
+        initialData={nodes[editIndex]}
+        onSubmit={data => {
+          editNode(editIndex, data)
+          setPanel('main')
+        }}
+      >
+        <button type="submit" className={styles.btn}>
+          {t('node.modify_node')}
+        </button>
+      </NodeEditForm>
+    </div>
+  )
+
+  const NodeMainPanel = () => (
+    <div className={styles.contentWrapper}>
+      <div className={styles.modalTitle}>
+        <p>{t('navbar.node')}</p>
+        <button type="button" onClick={onClose} className={styles.closeBtn}>
+          <img src={CloseIcon} alt="close icon" />
+        </button>
+      </div>
+
+      <div className={styles.switcher}>
+        <label htmlFor="node-connect-mode">{t('node.node_connect_mode')}</label>
+        <HelpTip title={t('node.node_connect_tooltip')} />
+        <Switch
+          id="node-connect-mode"
+          style={{ marginLeft: 'auto' }}
+          checked={isActivated}
+          onCheckedChange={checked => setIsActivated(checked)}
+        />
+      </div>
+
+      <div className={styles.nodeList}>
+        {nodes.map((n, index) => (
+          <div className={styles.node} key={n.url}>
+            <div className={styles.nodeTitle}>
+              <input onClick={() => switchNode(n.url)} type="checkbox" checked={nodeService.nodeEndpoint === n.url} />
```

Comment:
Is it guaranteed that `n.url` is unique for `checked={nodeService.nodeEndpoint === n.url}`.